specification/apicenter/data-plane/ApiCenter.DataApi/preview/2024-02-01-preview/azure.apicenter.dataapi.json

```diff
@@ -0,0 +1,1644 @@
+      "description": "Paged collection of Api items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The Api items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.Api"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiDefinition": {
+      "type": "object",
+      "description": "Paged collection of ApiDefinition items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiDefinition items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiDefinition"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiDeployment": {
+      "type": "object",
+      "description": "Paged collection of ApiDeployment items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiDeployment items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiDeployment"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedApiVersion": {
+      "type": "object",
+      "description": "Paged collection of ApiVersion items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The ApiVersion items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.ApiVersion"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedEnvironment": {
+      "type": "object",
+      "description": "Paged collection of Environment items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The Environment items on this page",
+          "items": {
+            "$ref": "#/definitions/Models.Environment"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Versions": {
```

Comment:
Where this entity (and especially enum values) are coming from? Or where is it referenced?

Reply:
One may use https://azure.github.io/typespec-azure/docs/emitters/typespec-autorest/reference/emitter#omit-unreachable-types

Reply:
To answer your question @azaslonov it's coming from the Versions enum which is defined in main.tsp and it's referenced by the @versioned tag in the same file: https://github.com/Bruce-Moe/azure-rest-api-specs/blob/brucemoe-apicenter-Microsoft.ApiCenter-2024-03-01-stable/specification/apicenter/DataApi/main.tsp

Do we want to follow the link above in order to prevent this type from being emitted in the spec?

Reply:
@azaslonov Following up on this comment before resolving.